images/fcos-bastion-image/root/usr/lib/systemd/system/dnsmasq-init.service

```diff
@@ -0,0 +1,22 @@
+[Unit]
+Description=Prepare dnsmasq configuration for dhcp service
+Wants=network-online.target
+After=network-online.target
+RequiresMountsFor=%t/containers
+ConditionPathExists=!/var/.dhcp-configured
+
+[Service]
+Environment=PODMAN_SYSTEMD_UNIT=%n
+EnvironmentFile=/usr/share/bastion_services/dhcp/dnsmasq_systemd.conf
+Restart=on-failure
+TimeoutStopSec=70
+ExecStartPre=/bin/rm -f %t/%n.ctr-id
+ExecStartPre=/bin/sh -c 'cp ${LOCAL_DNSMASQ_DIR}/dnsmasq.conf ${PODMAN_DNSMASQ_BASEDIR}/etc/'
```

Comment:
if we reboot the host, we would replace any previous state with the initial one.

This (and next line too) should go to the init script IMO.

Reply:
Wont `ConditionPathExists=!/var/.dhcp-configured` prevent from further execution on reboot?

Reply:
Just in case we need to execute the script independently, or to download latest fcos. As suggested below I'll make the fcos download as a separate service, let me know if I should move these 2 lines within the script.

Reply:
i think it's better to have those lines within the script. In particular, if you call it "dnsmasq-init", I'd expect all the `init` things to stay there.